### PR TITLE
Fix: Dropdown with "hover" trigger incorrectly rendered on mobile devices (#3961)

### LINKED ISSUE #3961: Dropdown with "hover" trigger incorrectly rendered on mobile devices
### Overview of the problem

**Buefy** version: v0.9.27
**Vuejs** version: 2.7.15
**OS/Browser**: MacOs/Chrome

### Description

The Dropdown does not render correctly in mobile view, when trigger 'hover' is set.
image

### Steps to reproduce

1. visit https://buefy.org/documentation/dropdown on smartphone or desktop browser mobile emulation
2. tap "Hover me!"
3. see incorrect position and no background

### Expected behavior

Behave like the "Click me!" tiggered button

### PR DESCRIPTION
**Pull request by kikuomax**
Fixes
- fixes #3961

## Proposed Changes

- Change the behavior of "hover" trigger on touch devices. It works like "click" trigger when the screen is narrow on touch devices. "hover" trigger won't change the behavior on non-touch devices, e.g., PC, even if the screen is narrow.
- The documentation page of `BDropdown` explains the new behavior of "hover" trigger.

### CODE REVIEW

**Comment by wesdevpro** [comment]
@kikuomax I am still having some weird behavior on my surface pro 7 (with and with out my monitors). The Hover button doesn't seem to trigger at all.

File: src/scss/components/_dropdown.scss
Comment (kikuomax) [inline src/scss/components/_dropdown.scss:45]: @wesdevpro Interesting. It might be due to 2-in-1. I conditionally applied the hover reaction to devices where the primary input method supports hover; e.g., a mouse.

Comment (kikuomax) [reply src/scss/components/_dropdown.scss:45]: @wesdevpro Let's what happens if I remove the condition.

Comment (wesdevpro) [reply src/scss/components/_dropdown.scss:45]: Okay👍🏼Let me know when you push your changes

**Comment by kikuomax** [comment]
> @kikuomax I am still having some weird behavior on my surface pro 7 (with and with out my monitors). The Hover button doesn't seem to trigger at all.

@wesdevpro Did you use your mouse, or touch? What if you touch the trigger?

**Comment by wesdevpro** [comment]
> > @kikuomax I am still having some weird behavior on my surface pro 7 (with and with out my monitors). The Hover button doesn't seem to trigger at all.
> 
> @wesdevpro Did you use your mouse, or touch? What if you touch the trigger?

Touch no longer works on my surface pro. The issue is caused by a loose cable from a bad repair that I did on the screen😅I used my mouse to both hover and click the "Hover me" button.

**Comment by wesdevpro** [comment]
@kikuomax the hover button is now working for me 
image

File: src/scss/components/_dropdown.scss
Comment (kikuomax) [reply src/scss/components/_dropdown.scss]: @wesdevpro Please try it out!

**Comment by kikuomax** [comment]
@wesdevpro We learned that Surface Pro believes itself a touch device even if a mouse is connected. Thanks for contributing to solving a subtle problem!

**Comment by kikuomax** [comment]
@wesdevpro Btw, please let me "Squash and merge" after you approve. Because I want to clean up the commit log.

**Comment by wesdevpro** [comment]
> @wesdevpro Btw, please let me "Squash and merge" after you approve this PR. Because I want to clean up the commit log.

Do you mean that you want to squash then force commit a single commit then merge the pull request?

**Comment by kikuomax** [comment]
> Do you mean that you want to squash then force commit a single commit then merge the pull request?

@wesdevpro I will do it on GitHub. I just want to edit this message:
image

**Comment by kikuomax** [comment]
@wesdevpro Please approve this PR if it looks good to you, anyway.

File: docs/pages/components/dropdown/Dropdown.vue
Comment (wesdevpro) [inline docs/pages/components/dropdown/Dropdown.vue:6]: Below are some minor grammatical changes that you can copy/paste👇🏼.
```
It appears as a modal for tablets and smartphones, more precisely when the screen is narrower than <code>$dropdown-mobile-breakpoint</code>. However, dropdowns with <code>"hover"</code> trigger won't change its behavior to avoid any malfunction with hover. <code>"hover"</code> trigger works like <code>"click"</code> trigger on touch devices where hover events do not make sense.<code>"hover"</code> trigger precedes <code>"click"</code> trigger.
```

**Review by wesdevpro — CHANGES_REQUESTED** [review]

**Review by wesdevpro — APPROVED** [review]
LGTM 🚈

**Pull request merged**
